components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -0,0 +1,610 @@
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.json.JSONObject;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.gateway.threatprotection.utils.ThreatProtectorConstants;
+import org.wso2.carbon.apimgt.gateway.utils.GatewayUtils;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.JsonPath;
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.config.Entry;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+
+/**
+ * This SchemaValidator handler validates the request/response messages against schema defined in the swagger.
+ */
+public class SchemaValidator extends AbstractHandler {
+
+    private static final Log logger = LogFactory.getLog(SchemaValidator.class);
+    private String uuid;
+    private String swagger = null;
+    private JsonNode rootNode;
+    private String requestMethod;
+    private String schemaContent = null;
+
+    public void setLocalentry(String localEntry) {
+        this.uuid = localEntry;
+    }
+
+    /**
+     * Handle the API request message validation.
+     *
+     * @param messageContext Context of the message to be validated
+     * @return Continue true the handler chain
+     */
+    @Override
+    public boolean handleRequest(MessageContext messageContext) {
+        if (logger.isDebugEnabled()) {
+            logger.info("Validating the API request Body content..");
+        }
+        org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext)
+                messageContext).getAxis2MessageContext();
+        String contentType;
+        Object objContentType = axis2MC.getProperty(ThreatProtectorConstants.REST_CONTENT_TYPE);
+        if (objContentType != null) {
+            contentType = objContentType.toString();
+            if (logger.isDebugEnabled()) {
+                logger.info("Content type of the request message :" + contentType);
+            }
+            try {
+                RelayUtils.buildMessage(axis2MC);
+                if (logger.isDebugEnabled()) {
+                    logger.info("Successfully built the request message");
+                }
+                if (ThreatProtectorConstants.APPLICATION_JSON.equals(contentType)) {
```

Comment:
As I see here also we can't continue if `contentType` is not `APPLICATION_JSON`. We should be able to switch the if block and return to keep the code and execution simple.